mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -43,10 +43,13 @@ import org.apache.spark.sql.DataFrame
 @Since("1.0.0")
 class BinaryClassificationMetrics @Since("1.3.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[(Double, Double)],
-    @Since("1.3.0") val numBins: Int) extends Logging {
+    @Since("1.3.0") var numBins: Int) extends Logging {
```

Comment:
It looks like a typo in the comment, and intends to refer to the calling class. See https://github.com/mpjlu/spark/commit/c5c6aded641048a3e66ac79d9e84d34e4b1abae7 The caller is setting a default to 100, not BinaryClassificationMetrics.

Why not add a method? because of the complexity it introduces -- exactly what your PR has to add. I would not make this change.